code/modules/mob/living/simple_animal/bot/secbot.dm

```diff
@@ -178,18 +226,83 @@ Auto Patrol: []"},
 		retaliate(H)
 		if(special_retaliate_after_attack(H))
 			return
+	if(H.a_intent == INTENT_HELP && bot_accessory)
+
+		to_chat(H, "<span class='warning'>You knock [bot_accessory] off of [src]'s head!</span>")
+		bot_accessory.forceMove(get_turf(src))
+		//reset all emotes/sounds and name/desc
```

Comment:
make the reset a proc too, please. 
also make applying the fashion be a proc that accepts a datum typepath